### PR TITLE
feat: deprecate `multiline-comment-style` & `line-comment-position`

### LINKED ISSUE #17681: Change Request: Consider also moving slightly-more-than-formatting rules out of core
### ESLint version

8.52.0

### What problem do you want to solve?

(Moving https://github.com/eslint/eslint/issues/17522#issuecomment-1775868616 here) Following up on rules such as `curly` and `padding-line-between-statements` (https://github.com/eslint-stylistic/eslint-stylistic/issues/26 -> https://github.com/eslint-stylistic/eslint-stylistic/issues/26#issuecomment-1773890620): there are some core rules that aren't quite covered by some or all formatters, but are still within the realm of what eslint-stylistic would cover. Could we also move those rules out of core, with the intent to move them to eslint-stylistic?

### What do you think is the correct solution?

Rules such as `curly` that are at or just above the formatting level suffer from many of the maintainability drawbacks mentioned in #17522.  The eslint-stylistic project is intentionally named _stylistic_ rather than _formatting_ so that it can encompass more than just formatting rules. See https://github.com/eslint-stylistic/eslint-stylistic/issues/26.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

Sorry for raising this so late after the initial discussion!

### PR DESCRIPTION
**Pull request by aladdin-add**
fixes #17681

#### Prerequisites checklist

- [x ] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x] Other, please explain:

#### Is there anything you'd like reviewers to focus on?
no.

### CODE REVIEW

File: docs/src/rules/line-comment-position.md
Comment (aladdin-add) [inline docs/src/rules/line-comment-position.md:6]: it's required by the checker. I was presuming `v9.3.0` is the upcoming version.

Comment (mdjermanovic) [reply docs/src/rules/line-comment-position.md:6]: Can we also add a link to the corresponding rule, like we are doing with other rules that are moved to the stylistic plugin? For example:

https://github.com/eslint/eslint/blob/a63ed722a64040d2be90f36e45f1f5060a9fe28e/docs/src/rules/array-bracket-newline.md?plain=1#L8

Comment (aladdin-add) [reply docs/src/rules/line-comment-position.md:6]: It doesn't exist yet: https://github.com/eslint-stylistic/eslint-stylistic/issues/388

I can add a link when it's done.

Comment (mdjermanovic) [reply docs/src/rules/line-comment-position.md:6]: Ah, the rules haven't been published yet?

https://github.com/eslint-stylistic/eslint-stylistic/pull/389

In that case, merging this PR will need to wait for the new version of the plugin.

Comment (antfu) [reply docs/src/rules/line-comment-position.md:6]: ESLint Stylistic v2.1 has been released with these rules

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Since this is a `feat`, it needs another review from a team member before merging.

**Review by Tanujkanti4441 — APPROVED** [review]
LGTM, thanks!

**Review by nzakas — APPROVED** [review]

**Pull request merged**
